mint-auditor/tests/integration_test.py

```diff
@@ -0,0 +1,270 @@
+
+        request = GenerateBurnRedemptionTxRequest(
+            sender_monitor_id=monitor_id,
+            change_subaddress=change_subaddress,
+            input_list=input_list,
+            burn_amount=burn_amount,
+            fee=fee,
+            tombstone=tombstone,
+            token_id=token_id,
+            redemption_memo=redemption_memo,
+            enable_destination_memo=enable_destination_memo,
+        )
+        return self.stub.GenerateBurnRedemptionTx(request).tx_proposal
+
+    def submit_tx(self, tx_proposal):
+        return self.stub.SubmitTx(
+            SubmitTxRequest(
+                tx_proposal=tx_proposal,
+            )
+        )
+
+    def wait_for_monitor_to_sync(self, monitor_id, poll_interval=0.1):
+        while True:
+            network_block_index = self.get_network_block_index()
+            status = self.stub.GetMonitorStatus(
+                GetMonitorStatusRequest(
+                    monitor_id=monitor_id
+                )
+            ).status
+            if status.next_block >= network_block_index:
+                break
+            time.sleep(poll_interval)
+
+
+class MintAuditorTest:
+    def __init__(self, args):
+        self.args = args
+        self.mobilecoind_client = MobilecoindClient(args.mobilecoind_addr)
+        self.mint_auditor_client = MintAuditorClient(args.mint_auditor_addr)
+
+    def run(self):
+        monitor_id = self.mobilecoind_client.new_random_monitor()
+
+        try:
+            b58_addr = self.mobilecoind_client.get_b58_public_address(monitor_id)
+            logging.info(f"Minting to {b58_addr}")
+            token_id = 1
+            mint_amount = 10000
+
+            # Get the network block height and wait for the mint auditor to catch up
+            response = self.wait_for_mint_auditor_to_sync()
+            previous_minted_amount = dict(response.block_audit_data.balance_map).get(token_id) or 0
+
+            # Mint tokens that go into the new wallet we generated
+            logging.info(f"Minting {mint_amount} tokens of token_id {token_id}")
+            output = subprocess.check_output(" ".join([
+                self.args.mint_client_bin,
+                "generate-and-submit-mint-tx",
+                f"--node {self.args.node_url}",
+                f"--signing-key {self.args.mint_signing_key}",
+                f"--recipient {b58_addr}",
+                f"--token-id {token_id}",
+                f"--amount {mint_amount}",
+            ]), shell=True)
+            logging.info(f"Mint client output: {output}")
+
+            # Give the network time to process the mint and then for mobilecoind and the mint auditor to catch up
+            self.wait_for_balance(monitor_id, token_id, mint_amount)
+
+            response = self.wait_for_mint_auditor_to_sync()
+
+            current_minted_amount = dict(response.block_audit_data.balance_map).get(token_id)
+            assert current_minted_amount == previous_minted_amount + mint_amount
```

Comment:
Seems useful to include the amounts in the assert 'message'
```suggestion
            assert current_minted_amount == previous_minted_amount + mint_amount, (current_minted_amount, previous_minted_amount, mint_amount)
```